packages/components/src/mobile/hr/index.native.js

```diff
@@ -0,0 +1,58 @@
+/**
+ * External dependencies
+ */
+import { View, Text } from 'react-native';
+
+/**
+ * Internal dependencies
+ */
+import styles from './style.scss';
+
+function HRLine( { color, lineStyle } ) {
+	const lineStyles = [
+		styles.hr__line,
+		lineStyle,
+		color && { backgroundColor: color },
+	];
+	return <View style={ lineStyles } />;
+}
+
+function HRText( { text, textStyle } ) {
+	const textStyles = [ styles.hr__text, textStyle ];
+	return <Text style={ textStyles }>{ text }</Text>;
+}
+
+function HR( {
+	color,
+	lineStyle,
+	marginLeft = 8,
+	marginRight = 8,
+	text,
+	textStyle,
+} ) {
+	const containerStyles = [
+		styles.hr__container,
+		{
+			marginLeft,
+			marginRight,
+		},
+	];
+
+	if ( ! text ) {
+		return (
+			<View style={ containerStyles }>
+				<HRLine color={ color } lineStyle={ lineStyle } />
+			</View>
+		);
+	}
+
+	return (
+		<View style={ containerStyles }>
+			<HRLine color={ color } lineStyle={ lineStyle } />
+			<HRText color={ color } text={ text } textStyle={ textStyle } />
```

Comment:
Likely not necessary, but I wondered if it makes sense to rely upon the `children` prop instead of a `text` prop to enable component composition.

```js
// Definition
function HorizontalRule({ children }) {
  // ...

  return (
    <>
      <HRLine />
      {children}
      <HRLine />
    </>
  );
}

// Usage
<HorizontalRule>
  <Text>Read More</Text>
</HorizontalRule>;
```

If desired, we could take it a step further simplifying the usage by applying the required text styles via a nested component. 

```js
// Definition
function HorizontalRuleText( { children, style } ) {
  const requiredStyles = { // ...required text styling...  };

  return (
    <Text style={{...requiredStyles, ...style}} >
      {children}
    </Text>
  );
}

HorizontalRule.Text = HorizontalRuleText;

// Usage
<HorizontalRule>
  <HorizontalRule.Text>Read More</HorizontalRule.Text>
</HorizontalRule>;
```